hibernate-core/src/main/java/org/hibernate/engine/spi/CollectionEntry.java

```diff
@@ -428,7 +452,7 @@ public void serialize(ObjectOutputStream oos) throws IOException {
 	 */
 	public static CollectionEntry deserialize(
 			ObjectInputStream ois,
-			SessionImplementor session) throws IOException, ClassNotFoundException {
+			@Nullable SessionImplementor session) throws IOException, ClassNotFoundException {
```

Comment:
```suggestion
			SessionImplementor session) throws IOException, ClassNotFoundException {
```

Reply:
Weird, don't know why i would have had to mark this as nullable ...